mp/src/game/client/neo/ui/neo_scoreboard.cpp

```diff
@@ -1,255 +1,774 @@
-	HFont hFallbackFont = scheme()->GetIScheme(GetScheme())->
-		GetFont("DefaultVerySmallFallBack", false);
-
-	m_pPlayerList->AddSection(0, "");
-	m_pPlayerList->SetSectionAlwaysVisible(0);
-	m_pPlayerList->AddColumnToSection(0, "name", "", 0, m_iNameWidth, hFallbackFont);
-	m_pPlayerList->AddColumnToSection(0, "rank", "Rank", 0, m_iRankWidth, hFallbackFont);
-	m_pPlayerList->AddColumnToSection(0, "class", "Class", 0, m_iClassWidth, hFallbackFont);
-	m_pPlayerList->AddColumnToSection(0, "xp", "XP", 0, m_iScoreWidth, hFallbackFont);
-	m_pPlayerList->AddColumnToSection(0, "deaths", "#PlayerDeath", 0, m_iDeathWidth, hFallbackFont);
-	m_pPlayerList->AddColumnToSection(0, "ping", "Ping", 0, m_iPingWidth, hFallbackFont);
-	m_pPlayerList->AddColumnToSection(0, "status", "Status", 0, m_iStatusWidth, hFallbackFont);
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: Adds a new section to the scoreboard (i.e the team header)
+//-----------------------------------------------------------------------------
 void CNEOScoreBoard::AddSection(int teamType, int teamNumber)
 {
-	HFont hFallbackFont = scheme()->GetIScheme(GetScheme())->
-		GetFont("DefaultVerySmallFallBack", false);
+	IGameResources *gr = GameResources();
 
-	const int sectionID = GetSectionFromTeamNumber(teamNumber);
-	m_pPlayerList->AddSection(sectionID, "", (teamNumber == TEAM_SPECTATOR) ? NULL : StaticPlayerSortFunc);
-	m_pPlayerList->SetSectionAlwaysVisible(sectionID);
+	if ( !gr )
+		return;
 
-	// setup the columns
-	if (teamType == TYPE_TEAM)
+	auto pPlayerList = GetPanelForTeam(teamNumber);
+	int sectionID = GetSectionFromTeamNumber(teamNumber);
+	if ( teamType == TYPE_TEAM )
 	{
-		if (ShowAvatars())
+		pPlayerList->AddSection(sectionID, "", StaticPlayerSortFunc);
+		pPlayerList->SetSectionAlwaysVisible(sectionID);
+
+		pPlayerList->AddColumnToSection(sectionID, "ping", "", 0, m_iPingWidth );
+		if ( ShowAvatars() )
 		{
-			m_pPlayerList->AddColumnToSection(sectionID, "avatar", "", SectionedListPanel::COLUMN_IMAGE, m_iAvatarWidth);
+			pPlayerList->AddColumnToSection( sectionID, "avatar", "", SectionedListPanel::COLUMN_IMAGE | SectionedListPanel::COLUMN_CENTER, m_iAvatarWidth );
 		}
-		m_pPlayerList->AddColumnToSection(sectionID, "name", "", 0, m_iNameWidth - m_iAvatarWidth, hFallbackFont );
-		m_pPlayerList->AddColumnToSection(sectionID, "rank", "", 0, m_iRankWidth, hFallbackFont);
-		m_pPlayerList->AddColumnToSection(sectionID, "class", "", 0, m_iClassWidth, hFallbackFont);
-		m_pPlayerList->AddColumnToSection(sectionID, "xp", "", 0, m_iScoreWidth, hFallbackFont);
-		m_pPlayerList->AddColumnToSection(sectionID, "deaths", "", 0, m_iDeathWidth, hFallbackFont);
-		m_pPlayerList->AddColumnToSection(sectionID, "ping", "", 0, m_iPingWidth, hFallbackFont);
-		m_pPlayerList->AddColumnToSection(sectionID, "status", "", 0, m_iStatusWidth, hFallbackFont);
+		pPlayerList->AddColumnToSection(sectionID, "name", "", 0, m_iNameWidth - m_iAvatarWidth );
```

Comment:
If we're not showing avatars, do we want to give names more space on the scoreboard?

Reply:
To be honest I'm not sure why the original code has this ShowAvatars method. Seems to be for some benchmarking purposes? In our case it'll always return true, but I'll leave it as is for now. Maybe in the future we'll want to disable them when activating the streamer mode.